pages/api/Schema/reports/expectedMonthlyTotal/datahandler.ts

```diff
@@ -0,0 +1,71 @@
+import { ExpectedMonthlyTotalReport } from '../../../graphql-rest.page.generated';
+
+export interface ExpectedMonthlyTotalResponse {
+  id: string;
+  type: string;
+  attributes: {
+    created_at: string;
+    expected_donations: {
+      contact_id: string;
+      contact_name: string;
+      contact_status: string;
+      converted_amount: number;
+      converted_currency: string;
+      converted_currency_symbol: string;
+      donation_amount: number;
+      donation_currency: string;
+      donation_currency_symbol: string;
+      pledge_amount: number;
+      pledge_currency: string;
+      pledge_currency_symbol: string;
+      pledge_frequency: string;
+      type: 'received' | 'likely' | 'unlikely';
+    }[];
+    total_currency: string;
+    total_currency_symbol: string;
+    updated_at: string | null;
+    updated_in_db_at: string | null;
+  };
+  relationships: {
+    account_list: {
+      data: {
+        id: string;
+        type: string;
+      };
+    };
+  };
+}
+
+const mapDonation = (
+  donation: ExpectedMonthlyTotalResponse['attributes']['expected_donations'][0],
```

Comment:
Would it help to have a separate interface for `expected_donations`?

Reply:
🤷‍♂️ We're doing this elsewhere with the GraphQL generated types... I guess it could be.

Reply:
It's not a big thing, whatever makes the most sense to you.

Reply:
I implemented that, let me know what you think.

Reply:
I think it's ok. I sorta like that it's one big type since it matches the JSON coming in and you can compare it all at once. 🤷‍♂️